### PR TITLE
colexec: remove unused Int8 and Float32 coltypes

### PR DESCRIPTION
**Pull request by yuzefovich**
We have two types (Int8 and Float32) which are not used at all, yet
the code is generated for them which inflates the footprint of the
generated code even further. Now those two types are removed which
reduces the binary size by 8% or so.

Int8 doesn't even have an equivalent types in sql/types, but Float32
does, and if we ever start using it, we can always bring it back later.

Release justification: Category 1: Non-production code changes.

Release note: None

### CODE REVIEW

**Review by rafiss — APPROVED** [review]
thanks for doing this!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/40831#-:-Lozwae56MxPVjZsRkmS:bkbzvqp)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @danhhz, @jordanlewis, and @yuzefovich)

___
*[pkg/sql/colexec/execgen/cmd/execgen/overloads.go, line 778 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/40831#-LozwMwK2Ag_PcDIS4dX:-LozwMwK2Ag_PcDIS4dY:b7ampnf) ([raw file](https://github.com/cockroachdb/cockroach/blob/6abf52aec95abe8f2de0eecbd5613e1595311999/pkg/sql/colexec/execgen/cmd/execgen/overloads.go#L778)):*
> ```Go
> 			var upperBound, lowerBound string
> 			switch c.width {
> 			case 8:
> ```

nit: this case can be removed

**Review by yuzefovich — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/40831)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @danhhz, @jordanlewis, and @rafiss)

___
*[pkg/sql/colexec/execgen/cmd/execgen/overloads.go, line 778 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/40831#-LozwMwK2Ag_PcDIS4dX:-Lp0ZjfB9hBvjeF7gXxf:bo3vl6l) ([raw file](https://github.com/cockroachdb/cockroach/blob/6abf52aec95abe8f2de0eecbd5613e1595311999/pkg/sql/colexec/execgen/cmd/execgen/overloads.go#L778)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

nit: this case can be removed
</blockquote></details>

Thanks, done!

**Comment by yuzefovich** [comment]
I think I should get an LGTM from @jordanlewis or @asubiotto before merging this.

**Review by jordanlewis — APPROVED** [review]

**Comment by yuzefovich** [comment]
Thanks for the reviews!

bors r=rafiss,jordanlewis

**Pull request merged**
